### ISSUE TITLE
 UserWarning during training with BPTT

### ISSUE THREAD

**Issue by ANYMS-A** [synsense/sinabs#96]
warn message is something like below:
```
//venv/lib/python3.8/site-packages/torch/autograd/__init__.py:173: UserWarning: An output with one or more elements was resized since it had shape [], which does not match the required output shape [batch_size, num_classes]. This behavior is deprecated, and in a future PyTorch release outputs will not be resized unless they have zero elements. You can explicitly reuse an out tensor t by resizing it, inplace, to zero elements with t.resize_(0). (Triggered internally at  ../aten/src/ATen/native/Resize.cpp:17.)
  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
```

Though it doesn't efftect the usage, I am curious that how does this warning come?

**Comment by biphasic** [comment]
Interesting. What version of PyTorch are you using? This error comes from a `resize` call but Sinabs doesn't contain any resizing. I'm thinking that a required shape of [batch size, num_classes] is normally the case for a loss criterion such as cross entropy of MSE. Could that be it?

**Comment by ANYMS-A** [comment]
> Interesting. What version of PyTorch are you using? This error comes from a `resize` call but Sinabs doesn't contain any resizing. I'm thinking that a required shape of [batch size, num_classes] is normally the case for a loss criterion such as cross entropy of MSE. Could that be it?

Ah I think maybe the problem can be with the loss function! Thanks for the hint!

**Comment by biphasic** [comment]
Closing this for now, please reopen if this is still an issue
